### PR TITLE
RR-1464 - Schedule pending Induction on prison admission when S&As complete (PES)

### PR DESCRIPTION
**Pull request by ayoMoj**
Handles the prison-admission side of the PES rule for a prisoner who already has
an Induction Schedule awaiting their Screening & Assessments
(PENDING_INITIAL_SCREENING_AND_ASSESSMENTS_FROM_CURIOUS). On admission, if their
S&As are now complete (an "all relevant assessments complete" record exists), the
Induction is scheduled (today + 10 days); otherwise it is left pending to await the
Curious "S&As completed" message.

Reuses InductionScheduleService.schedulePendingInductionSchedule (added previously)
and adds EducationAssessmentEventService.hasCompletedAllAssessments. Dormant under
PEF: only PES ever produces PENDING schedules, so the new branch never fires in
prod until the PES rule is enabled.

### CODE REVIEW

File: src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/messaging/PrisonerReceivedDueToAdmissionEventPesTest.kt [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      domainEventQueueClient.countMessagesOnQueue(domainEventQueue.queueUrl).get()
+    } matches { it == 0 }
+
+    await untilAsserted {
+      val inductionSchedule = getInductionSchedule(prisonNumber)
+      assertThat(inductionSchedule)
+        .wasStatus(InductionScheduleStatusResponse.SCHEDULED)
+        .hasDeadlineDate(LocalDate.now().plusDays(10))
+    }
+  }
```
Comment (nathanrussell-moj-digital) [inline src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/messaging/PrisonerReceivedDueToAdmissionEventPesTest.kt:101]: Great test 👍. This is essentially the test for one of the edge case scenarios (ie. they are admitted to prison, but somehow they already have their S&A's done before they entered prison.
It's the right test for that scenario 👍 

We'll add more tests for some of the other scenarios as you work through this ticket, so this is great that we have this to move forward with 👍

File: src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/service/EducationAssessmentEventService.kt [hunk 2]
```diff
@@ -43,6 +44,15 @@ class EducationAssessmentEventService(
     log.info { "Saved education assessment event [${entity.reference}] for prisoner [$prisonNumber]" }
   }
 
+  /**
+   * Returns true if the prisoner has completed all of their relevant Education Screening & Assessments (i.e. we have
+   * received an "all relevant assessments complete" event for them from the external assessment system).
+   */
+  fun hasCompletedAllAssessments(prisonNumber: String): Boolean = educationAssessmentEventRepository.existsByPrisonNumberAndStatus(
```
Comment (nathanrussell-moj-digital) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/service/EducationAssessmentEventService.kt:51]: Nice 👍 🙌

**Review by nathanrussell-moj-digital — APPROVED** [review]
Love it; this is great 👍 
I love it when we can see through the problem and requirements, and can work out how to break it down and build out the functionality iteratively through a series of PRs.
Well done 👍
